### ISSUE TITLE
Collection of ideas for the shiny app

### ISSUE THREAD

**Issue by markomi** [SwissClinicalTrialOrganisation/secuTrialR#164]
I've written down a few ideas for the next version of the shiny app. How to make it more user-friendly and appealing for the user and amore robust.

- upload: once data is loaded, show the information permanently, e.g. in a single line on the upload tab, but also at the top of the page in other tabs. 
- upload: add a button that "unloads" loaded data
- upload error handling: seems okay in the upload tab, but errors are not handled well in other tabs.
- form completeness: "percent" are not shown as percent but as ratios. the labeling is thus misleading
- form completeness: I find it a bit misleading that the percent switch creates a separate table with % data. Also contributing to the confusion is that the two tables are not labeled. Might be better to make a single table with all info, which one can add or remove. 
- make a new module providing an option to download content of all tabs (incl tables/plots) as a pdf report. Here's the general approach: https://shiny.rstudio.com/articles/generating-reports.html
- visit plan: provide a text explaining the graph. One could assemble a paragraph based on the data frame used to make the plot, and then display it above/below the plot.
- design: unify the colouring scheme (e.g. the navbar and sidebar-toggle has a different colour than the header)
- design: make all buttons/input boxes/toggles on the same tab page same width & aligned. It will look a bit cleaner.
- recruitment plot: plot is not resizing properly
- maybe add a tab/module that allows the user to subset data. Not 100% sure if really necessary.
- stata-sas-spss download tab: there might be too many buttons here. Try implementing a dropdown menu for conversion type selection + download button
- monitoring cases: cannot select a center with example data? Always "all" centers are selected. Check how this can be fixed

**Comment by aghaynes** [comment]
> upload: once data is loaded, show the information permanently, e.g. in a single line on the upload tab, but also at the top of the page in other tabs.

How about something in the menu on the left instead?
![image](https://user-images.githubusercontent.com/6594028/75982367-c4796300-5ee6-11ea-90b8-80604b2694e1.png)
No idea if its possible, but then it's evident without needing to have a sentence on each page... (inless someone closes the panel)

**Comment by aghaynes** [comment]
> upload error handling: seems okay in the upload tab, but errors are not handled well in other tabs.

In the presize app, I do something like this
```
if(is.na(input$mean_n) & is.na(input$mean_ciwidth)) {
                cat("Awaiting 'number of observations' or 'confidence interval width'")
        } else {
```
The app then shows the message rather than an error. 
Just an idea :)

**Comment by markomi** [comment]
Yeah, maybe. But personally, I'd prefer knowing exactly what is loaded without having to go back to the Upload tab. Doesn't have to be a full sentence. Just sth like:
Loaded data: {data-file-name}

**Comment by aghaynes** [comment]
that could still go in the menu bar, perhaps not next to the upload button in that case, but just beneath it...?

**Comment by markomi** [comment]
Yeah maybe. I could try out a few different things and see where it looks best and is both informative and not distracting.

**Comment by PatrickRWright** [comment]
I think a small "signature" type of "thing" at the bottom of every page would serve this purpose quite well.

So basically what @markomi just not top but rather bottom ;)

**Comment by markomi** [comment]
> > upload error handling: seems okay in the upload tab, but errors are not handled well in other tabs.
> 
> In the presize app, I do something like this
> 
> ```
> if(is.na(input$mean_n) & is.na(input$mean_ciwidth)) {
>                 cat("Awaiting 'number of observations' or 'confidence interval width'")
>         } else {
> ```
> 
> The app then shows the message rather than an error.
> Just an idea :)

That's a good hint. Also there's this option: https://shiny.rstudio.com/articles/validation.html

I've used it before in another app and it works well and looks very decent.

**Comment by markomi** [comment]
@aghaynes @PatrickRWright 
As discusssed, I have created the [secuTrialRshiny](https://github.com/SwissClinicalTrialOrganisation/secuTrialRshiny) package. All further development of and issue tracking for the shiny app should take place there. You are both listed as authors.

I have referred to this issue in [secuTrialRshiny issue 3](https://github.com/SwissClinicalTrialOrganisation/secuTrialRshiny/issues/3) and am thus closing #164.
